internal/pkg/cli/dialog/steps.go

```diff
@@ -0,0 +1,205 @@
+package dialog
+
+import (
+	"bufio"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/umisama/go-regexpcache"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/prompt"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/input"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/strhelper"
+)
+
+// stepsDialog to define steps and steps groups.
+type stepsDialog struct {
+	prompt prompt.Prompt
+}
+
+func newStepsDialog(prompt prompt.Prompt) *stepsDialog {
+	return &stepsDialog{prompt: prompt}
+}
+
+func (d *stepsDialog) ask() (input.StepsGroups, error) {
```

Comment:
Asi tam nie su osetrene duplicitne `step.id`.

![image](https://user-images.githubusercontent.com/19371734/160806689-033462b0-584a-4fb4-a9ec-00b11503a6bf.png)

![image](https://user-images.githubusercontent.com/19371734/160806818-8b8bef07-f165-4f4a-818c-0f0bea2810fd.png)

Reply:
Dalo by sa to vyriesit nejakou pomocnou `map`.

Reply:
+ mal by sa validovat aj `inputs.jsonnet`, aby tam tiez nemohli byt duplicitne `step ID`.
Kludne len nejaky `for` po nacitani suboru.

![image](https://user-images.githubusercontent.com/19371734/160809904-41a42d05-cb4f-494f-a74c-ffe37c5d1c84.png)

Reply:
Alternativa je, nemam tam `step.id`, ... staci ti hierarchia, nie je nutne (alebo na nieco ano?).
Ak budeme niekde v API `stepId` potrebovat tak ho mozeme generovat, napr. `group0-step1`.
Je to podobne ako `input.step` ale o uroven vyssie.
Tj. tiez by to slo vyriesit pomocou map.

Reply:
Jo počkat, ale zkomplikovalo by se tím vlastně přiřazování inputů ke stepům (https://github.com/keboola/keboola-as-code/blob/787063d635d353fa963bbff8e73a25c444c4d15e/internal/pkg/cli/dialog/inputs_detail.go#L254-L254). 🤔 Tak by se tam zadávala i `stepGroup` a do obou by se psalo pořadí (tj. index+1), nebo už je to zase moc velká komplikace a mám nechat raději to idčko?

Reply:
To by malo ist vyriesit pomocou `orderedMap`, `n-ty` key je `stepId`, pre `n-ty` step.

Reply:
Ale mozeme to mergnut aj bez toho, a upravit to neskor, ak ti to pride lepsie.

Reply:
Teď nevím jestli rozumím. 🤔  Když by tvůrce šablony vydefinoval třeba tyto stepy:

```
## Group

### Step

### Step

## Group

### Step
```

Tak v `inputs_detail` by se to preview vypsalo takto:

```
- Group 1
  - Step 1
  - Step 2
- Group 2
  - Step 3
```

A pokud by chtěl uživatel input přiřadit pod první step druhé groupy, tak by zadal `3`?

Reply:
btw. jak tak na to preview steps v `inputs_detail` koukám, tak bych do něj přidal i description.

Reply:
Nemyslel som to tak, ... v dialogu by som to `stepId` nechal, ale iba na pospajanie `input->step`.
Ale naopak v `inputs.jsonnet` nema ziadny zmysel.

Reply:
Ak chces, mozeme si zavolat nad kodom.

Reply:
Ahá, už rozumím. 👍

Reply:
Mno, tak jsem to nějak vybojoval, ale nejsem si jistej jestli se mi to líbí. 😅  https://github.com/keboola/keboola-as-code/pull/610/commits/3c2a8ebd77b18ea2ae92f72c99765f6f6628ce2a